Migrate ThumbnailImage.js to function component

kadiealexander opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/16210
PROPOSAL: GH_LINK_ISSUE(COMMENT)


### Tests

1. Navigate to a chat in New Expensify and upload an image attachment/navigate to a chat with an existing image uploaded
2. Confirm that the thumbnail that appears in Expensify looks normal and is a normal size. 

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above. 

### QA Steps

1. Navigate to a chat in New Expensify and upload an image attachment/navigate to a chat with an existing image uploaded
2. Confirm that the thumbnail that appears in Expensify looks normal and is a normal size. 

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

![image](https://github.com/Expensify/App/assets/59587260/e48d5f6a-978d-4422-9cf5-d3d192fd90fc)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

![image](https://github.com/Expensify/App/assets/59587260/28abd385-7318-4e56-bc7c-7f150851ae4e)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![image](https://github.com/Expensify/App/assets/59587260/a34c8b54-6da2-40be-9904-88aa084358eb)

</details>

<details>
<summary>Desktop</summary>

![image](https://github.com/Expensify/App/assets/59587260/9d70305d-2737-4e86-88ae-f7358f770f32)

</details>

<details>
<summary>iOS</summary>

![image](https://github.com/Expensify/App/assets/59587260/97b28aa7-c22f-448c-86e4-f15298d0b1a3)

</details>

<details>
<summary>Android</summary>

![image](https://github.com/Expensify/App/assets/59587260/c162564d-f5db-4bf1-a72e-f4c701c7471d)

</details>

kadiealexander committed:
pushing a draft PR to change component to function for thumbnail images

kadiealexander committed:
syntax changes and debugging completed with puneet

kadiealexander committed:
cleaned up code with prettier

kadiealexander committed:
added display name

kadiealexander committed:
changes to clean up the lint failures

puneetlath commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +33,68 @@ const defaultProps = {
+            <View style={[StyleUtils.getWidthAndHeightStyle(imageWidth, imageHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
+                <ImageWithSizeCalculation
+                    url={props.previewSourceURL}
+                    onMeasure={() => updateImageSize()}
```
I think we need to include the arguments in the arrow function. 
```suggestion
                    onMeasure={(width, height) => updateImageSize(width, height)}
```

kadiealexander committed:
ran prettier to tidy code changes

kadiealexander committed:
Update src/components/ThumbnailImage.js

Co-authored-by: Puneet Lath <puneet@expensify.com>

kadiealexander commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +33,67 @@ const defaultProps = {
+            <View style={[StyleUtils.getWidthAndHeightStyle(imageWidth, imageHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
+                <ImageWithSizeCalculation
+                    url={props.previewSourceURL}
+                    onMeasure={(width, height) => updateImageSize(width, height)}
```
On Tuesday I'd love for you to show me how you found out more about this function, I've googled it to try understand what it's doing and what needs to be passed and I'm coming up a bit blank!

kadiealexander committed:
Updating the arguments for the onMeasure function

Updating the arguments for the onMeasure function

Co-authored-by: Puneet Lath <puneet@expensify.com>

kadiealexander commented:
@puneetlath this is now ready for review! Sorry for the delays, ran into a few ZeroTrust related bugs with my dev environment.

s77rt commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +33,67 @@ const defaultProps = {
+            <View style={[StyleUtils.getWidthAndHeightStyle(imageWidth, imageHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
+                <ImageWithSizeCalculation
+                    url={props.previewSourceURL}
+                    onMeasure={(measurements) => updateImageSize(measurements)}
```
```suggestion
                    onMeasure={updateImageSize}
```

Will need to get this back as it was

s77rt commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +33,67 @@ const defaultProps = {
 ThumbnailImage.propTypes = propTypes;
 ThumbnailImage.defaultProps = defaultProps;
+ThumbnailImage.displayName = 'ThumbnailImage';
 export default withWindowDimensions(ThumbnailImage);
```
Let's use `useWindowDimensions` hook instead

s77rt replied:
Given that we used `PureComponent` before we use use `React.memo` here.

kadiealexander committed:
update updateImageSize to use callback function

kadiealexander committed:
update onMeasure to pass updateImageSize function directly

kadiealexander committed:
use useWindowDimensions hook instead of HOC prop

kadiealexander committed:
update variables to factor in undefined states

s77rt commented:
@kadiealexander Lint please

kadiealexander committed:
fix lint

kadiealexander committed:
fix prettier

kadiealexander commented:
Sorry @s77rt, I've fixed the lint issues now and retested on web, just to check! 

![image](https://github.com/Expensify/App/assets/59587260/63a690c1-3530-41ea-bb52-557b05fb113b)

s77rt commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +31,71 @@ const defaultProps = {
+    );
+    return (
+        <View style={[props.style, styles.overflowHidden]}>
+            <View style={[StyleUtils.getWidthAndHeightStyle(imageWidth || 200, imageHeight), styles.alignItemsCenter, styles.justifyContentCenter]}>
```
Why we have this fallback `imageWidth || 200`? It was not there before. Your commit indicates it's about some undefined state case, can you elaborate on that?

kadiealexander replied:
This was included to resolve the following error that was occurring with `imageWidth`:

```
Argument of type 'number | undefined' is not assignable to parameter of type 'number'. Type 'undefined' is not assignable to type 'number'
```

s77rt replied:
This is a bug on `main` as well right? Let's either fix the root cause, I assume this is coming from `calculateThumbnailImageSize` where it returns an empty object if width or height is not truthy. Or work on this separately in a new issue.

kadiealexander replied:
Thansk for that! I've updated the `calculateThumbnailImageSize` function to return a value instead of being empty.

s77rt replied:
@kadiealexander Thanks for the update! Can you please include the steps to reproduce the bug? I don't see any errors from my end

puneetlath replied:
I believe it was just a lint error, not a bug. The linter was complaining that `imageWidth` might not be set.

s77rt replied:
Ah okay. I think we can move on with a default 200x200. BTW, is there any real case where the width or height is not set? I thought this would be the case with svg files but those does not have a thumbnail preview

kadiealexander replied:
I don't think so? We upload SVGs as files, which don't have a thumbnail. I feel like all other image types should have a fixed height and width.

MonilBhavsar replied:
So there is no way that width and height could be undefined in `calculateThumbnailImageSize()`, right?
If so, can we remove that?

kadiealexander replied:
@MonilBhavsar we're not bothered about the lint error in this instance?

MonilBhavsar replied:
I think linter error was there because we were returning an empty object, and then trying to access the offset.
We were returning empty object if either of width or height is undefined. I feel like width and height of image can never be undefined and if we can skip the check completely - 
```js
if (!width || !height) {
    return {thumbnailWidth: 200, thumbnailHeight: 200};
}
```
But, NAB 👍

kadiealexander committed:
fix potential undefined value error

kadiealexander committed:
make it prettier

s77rt commented:
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
![web](https://github.com/Expensify/App/assets/16493223/c044e8a7-42fb-49be-abb9-cbceca761a09)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
![mweb-chrome](https://github.com/Expensify/App/assets/16493223/8b48f9ec-86a8-40c7-9246-8bf7d22f45a3)

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
![mweb-safari](https://github.com/Expensify/App/assets/16493223/5205e795-78d7-4e29-969f-6eccdf6765a7)

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
![desktop](https://github.com/Expensify/App/assets/16493223/1bf5fba9-fbc5-4402-9eb8-796244a0f066)

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
![ios](https://github.com/Expensify/App/assets/16493223/383816d7-d3a3-4628-b31d-73b1e74c0f1f)

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
![android](https://github.com/Expensify/App/assets/16493223/92671429-51d5-4451-8e45-84ff2759cccb)

</details>

s77rt reviewed (approved):
Looks Bliss to me :rocket:

MonilBhavsar commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +31,71 @@ const defaultProps = {
+ * Compute the thumbnails width and height given original image dimensions.
+ *
+ * @param {Number} width - Width of the original image.
+ * @param {Number} height - Height of the original image.
```
NAB, Let's also describe third param - windowHeight?

s77rt replied:
Did we disable lint for JSDoc? Those kind of issues were being caught by the linter

MonilBhavsar replied:
Yeah, I think so. I haven't seen any error for not adding param JSDoc.
Anyways, NAB. But it think it would be good to have

MonilBhavsar commented on src/components/ThumbnailImage.js:
```diff
@@ -33,73 +31,71 @@ const defaultProps = {
-        };
+function calculateThumbnailImageSize(width, height, windowHeight) {
+    if (!width || !height) {
+        return {thumbnailWidth: 200, thumbnailHeight: 200};
```
Hmm, why we're returning this object now, instead of empty object previously?

s77rt replied:
https://github.com/Expensify/App/pull/20892#discussion_r1262114113

MonilBhavsar replied:
Thanks! I commented there https://github.com/Expensify/App/pull/20892#discussion_r1274902832

MonilBhavsar reviewed (dismissed):
Great work @kadiealexander!! Really appreciate it 👏 🚀 
Feel free to self merge after resolving the comment

MonilBhavsar commented:
Oops, looks like we have some conflicts to resolve 😬

kadiealexander committed:
Merge branch 'main' into kadie-thumbnail-component-change

kadiealexander committed:
adding new function argument to JSDocs description

kadiealexander committed:
adding new JSDocs parameter for new function argument

kadiealexander committed:
remove check for invalid image height/width since we'll always pass an image

kadiealexander commented:
@MonilBhavsar I've removed the check as discussed [here](https://github.com/Expensify/App/pull/20892#discussion_r1276129696), and added JSDoc param as discussed [here](https://github.com/Expensify/App/pull/20892#discussion_r1274873955).  Please review when you get a mo!

MonilBhavsar reviewed (commented):
I left a comment here to cross verify any edge case we're missing by removing that check https://github.com/Expensify/App/pull/9346#discussion_r1278862036

s77rt commented:
Can we move forward with this PR?

kadiealexander commented:
Response from [thread](https://github.com/Expensify/App/pull/9346#discussion_r1285368306):

>Apologies for the late reply. I don't remember exactly why this was added but it looks to me that as long as we're making sure that height and width are not undefined, this won't be an issue.

In that case I think we're good to proceed @MonilBhavsar?

MonilBhavsar commented:
@puneetlath all yours!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5825488105) to staging by https://github.com/puneetlath in version: 1.3.53-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5859809004) to production by https://github.com/yuwenmemon in version: 1.3.53-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
